core/server/worker/src/main/java/alluxio/worker/block/allocator/GreedyAllocator.java

```diff
@@ -73,6 +73,20 @@ private StorageDirView allocateBlock(long sessionId, long blockSize,
       return null;
     }
 
+    String mediumType = location.mediumType();
+    if (!mediumType.equals(BlockStoreLocation.ANY_MEDIUM)
+        && location.equals(BlockStoreLocation.anyDirInTierWithMedium(mediumType))) {
+      for (StorageTierView tierView : mManagerView.getTierViews()) {
+        for (StorageDirView dirView : tierView.getDirViews()) {
+          if (dirView.getMediumType().equals(mediumType)
+              && dirView.getAvailableBytes() >= blockSize) {
+            return dirView;
+          }
+        }
+      }
+      return null;
+    }
```

Comment:
Do we need to check the medium type in the first loop?

Reply:
first loop? outer loop?

Reply:
Nevermind, I was mistaken about the definition of `anyTier`